### ISSUE TITLE
String substitution adds extra space

### ISSUE THREAD

**Issue by alexey-terentiev** [chimpler/pyhocon#55]
Config file test.conf:
```
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
```
Code:
```
from pyhocon import ConfigFactory
config = ConfigFactory.parse_file("test.conf")
print(config['app']['java_opts'])
```
Result: `['-Xms128 m', '-Xmx128 m']`
Expected result: `['-Xms128m', '-Xmx128m']`

**Comment by alexey-terentiev** [comment]
In this case there is a workaround to have `app.heap_size = 128m` instead of `app.heap_size = 128`. But I guess that wont be an universal solution.

**Comment by darthbear** [comment]
Thank you @alexey-terentiev for the detailed bug report. It's been implemented in `pyhocon==0.3.15`.
